Updated header styling for subpages

hn4733 opened the pull request:
The following has been done: 
1. Styled the header for the subpages. 
2. Homepage has site identity (site title + blue ring) without styled menu items.
3. Optimised the header for mobile view -> check subpage header.
4. Styling of header added in /src/styles/header-styles.js

hn4733 committed:
initial commit

fridaland commented on .codesandbox/workspace.json:
```diff
@@ -0,0 +1,41 @@
+{
```
Interesting, do we need to commit this file?

hn4733 replied:
No worries :) happy to help. I have removed this file as part of the new review. Please check the latest updates.

fridaland commented on src/components/molecules/navigation.js:
```diff
@@ -21,16 +21,15 @@ export default function Navigation() {
-          </Styled.a>
-        </li>
-      ))}
+      <ul>
```
Nice!

fridaland commented on src/components/organisms/navbar.js:
```diff
@@ -1,19 +1,39 @@
 
 NavBar.propTypes = {
-  siteTitle: PropTypes.string,
+  siteTitle: PropTypes.string
```
We are in favor of trailing commas here :) 
```suggestion
  siteTitle: PropTypes.string,
```

fridaland commented on src/components/organisms/navbar.js:
```diff
@@ -1,19 +1,39 @@
-)
+const NavBar = ({ isHome }) => {
+  const [hamburgerState, setHamburgerState] = useState(false)
+  console.log({ hamburgerState })
```
Thoughts on removing this `console.log`?

hn4733 replied:
I have removed the console log in the review file now. :)

fridaland reviewed (changes requested):
This is awesome! Thanks for your contribution! Just have a couple minor thoughts/comments.

hn4733 committed:
Delete workspace.json

We don't need this file. It must have been pushed with the commit from codesandbox.io

hn4733 committed:
Updated navbar.js

1. Removed console.log for the hamburger state.
2. Added type checking for the "isHome" prop.

hn4733 committed:
Updated layout.js

1. Added type checking for the "isHome" prop.

hn4733 reviewed (commented):
Updated the navBar.js, and layout.js files
1. Type checking added for the "isHome" prop.
2. Removed console.log in navBar.js
3. Added trailing comma for the propTypes.
4. Removed the codesandbox workspace file (pushed with the commit from codesandbox.io)

fridaland commented on src/components/organisms/navbar.js:
```diff
@@ -1,19 +1,41 @@
 
 NavBar.propTypes = {
   siteTitle: PropTypes.string,
+  isHome: PropTypes.Bool,
```
There are a couple issues with the build on this Proptype, looks like it just needed to be lowercase!
```suggestion
  isHome: PropTypes.bool,
```

fridaland commented on src/components/templates/layout.js:
```diff
@@ -14,9 +14,15 @@ const Layout = () => {
+}
+  
+Layout.propTypes = {
+  isHome: PropTypes.Bool,
```
```suggestion
  isHome: PropTypes.bool,
```
Also looks like we forgot to import : 
```
import PropTypes from "prop-types"
```

fridaland reviewed (changes requested):
A few minor comments after checking this out locally

zopa commented:
One thing --- the big In The Fight in the header shouldn't wrap. ie, 
![image](https://user-images.githubusercontent.com/1517472/97750688-ac82cf00-1ac7-11eb-9b0d-b1497c3b36f5.png)

rather than

![header-wrap](https://user-images.githubusercontent.com/1517472/97750795-dd630400-1ac7-11eb-8643-c86e6b4905f4.jpg)

hn4733 committed:
Update src/components/organisms/navbar.js

Co-authored-by: Frida Casas <fridacasas.fc@gmail.com>

hn4733 committed:
Update layout.js

hn4733 committed:
Merge branch 'master' into csb-13mj6

hn4733 committed:
Minor adjustments
1. Adjusted import quote declaration from double to single for consistency.

hn4733 committed:
Delete workspace.json

hn4733 commented:
I've reviewed your feedback and adjusted accordingly. Also adjusted doubles quotes to single for the import statements in navbar.js and layout.js. Tested everything to work correctly. However, looking at the console logs, I have noticed that there's quite a few errors that have not been addressed ie. no key values placed in loops and some data that are undefined. You may need to create issues for those. I'm not sure what @zopa is referring to but the header looks correct on all subpages - homepage header was originally kept with no styling - brief was to only style the **subpage** header.

fridaland commented:
@hn4733 Great! Thanks for those updates, I believe what @zopa was referring to is that the actual `IN THE FIGHT` words should not span to two lines. The screenshot above was taken when we checked out the branch locally, so to adjust we would just have the title span one line and the logo sized appropriately next to it. Let me know if this makes sense and if you are able to adjust!

_***Edit:_ looks like the style issue with the title is sorted out with the updates!

 Please make sure to run `eslint` and resolve all lints now that we have incorporated it into our CI. :)

fridaland reviewed (approved):
Looks like those updates sorted out the issues with the logo ✨ 
Thanks for your contribution! As soon as CI passes (looks like just a few minor lints) this is good to go!

Tip: some of these may be resolved by running `eslint --fix src`
To run the linter: `eslint src`

fridaland committed:
Lints

fridaland commented:
Took care of those lints for ya. Great job @hn4733!

hn4733 commented:
> Took care of those lints for ya. Great job @hn4733!

Thanks :) Glad I could contribute!
